discovery-infra/test_infra/utils/utils.py

```diff
@@ -814,6 +814,41 @@ def fetch_url(url, timeout=60, max_retries=5):
     except (RequestException, HTTPError) as err:
         raise Exception(f"Failed to GET: {url} ({err})")
 
+def are_agents_in_status(
```

Comment:
IMO this logic should be under the controller as a static method.
WDYT? @tsorya  @eliorerz

Reply:
@YuviGold Agree

Reply:
Sure, may I know which controller is best suited for this?

Reply:
Agent